diff --git a/CMakeLists.txt b/CMakeLists.txt
index f0f99dc..99f9db7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,7 @@ include(CMakePackageConfigHelpers)
 
 enable_testing()
 
-add_definitions(-D_CRT_SECURE_NO_WARNINGS -g -Weverything -Wpedantic -Werror -Wno-padded -Wno-c++98-compat -Wno-documentation -std=c++11)
+add_definitions(-D_CRT_SECURE_NO_WARNINGS -g -Wpedantic -Werror -Wno-padded -Wno-c++98-compat -Wno-documentation -Wno-undefined-reinterpret-cast -Wno-c++98-compat-pedantic --std=c++11)
 
 option(GLM_STATIC_LIBRARY_ENABLE "GLM static library" OFF)
 if(GLM_STATIC_LIBRARY_ENABLE)
